src/persist-client/src/internal/state.rs

```diff
@@ -1294,15 +1311,17 @@ where
         key_schema: &K::Schema,
         val_schema: &V::Schema,
     ) -> ControlFlow<NoOpStateTransition<Option<SchemaId>>, Option<SchemaId>> {
-        fn data_type<T>(schema: &impl Schema2<T>) -> Bytes {
-            // To be defensive, create an empty batch and inspect the resulting
-            // data type (as opposed to something like allowing the `Schema2` to
-            // declare the DataType).
-            let array = Schema2::encoder(schema).expect("valid schema").finish();
-            let proto = Array::data_type(&array).into_proto();
+        fn encoded_data_type(data_type: &DataType) -> Bytes {
+            let proto = data_type.into_proto();
             prost::Message::encode_to_vec(&proto).into()
         }
 
+        fn decode_data_type(buf: Bytes) -> Result<DataType, anyhow::Error> {
```

Comment:
Tiny nit, but the names here aren't quite consistent: `encoded` vs. `decode`.

Reply:
Fixed!